gossip/scrambler_new/tx_scrambler_test.go

```diff
@@ -0,0 +1,544 @@
+	entries := []ScramblerEntry{
+		&dummyScramblerEntry{
+			hash:   common.Hash{1},
+			sender: common.Address{1},
+			nonce:  1,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{1},
+			sender: common.Address{1},
+			nonce:  1,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{1},
+			sender: common.Address{1},
+			nonce:  1,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{1},
+			sender: common.Address{1},
+			nonce:  1,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{1},
+			sender: common.Address{1},
+			nonce:  1,
+		},
+	}
+
+	shuffleEntries(entries)
+	entries = scrambleTransactions(42, entries)
+	seen := map[common.Hash]struct{}{}
+	for _, entry := range entries {
+		if _, ok := seen[entry.Hash()]; ok {
+			t.Fatalf("duplicate entry found %s", entry.Hash().Hex())
+		}
+		seen[entry.Hash()] = struct{}{}
+	}
+}
+
+func TestTxScrambler_SortTransactionsWithSameSender_SortsByNonce(t *testing.T) {
+	entries := []ScramblerEntry{
+		&dummyScramblerEntry{
+			hash:   common.Hash{1},
+			sender: common.Address{1},
+			nonce:  1,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{2},
+			sender: common.Address{1},
+			nonce:  2,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{3},
+			sender: common.Address{1},
+			nonce:  3,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{4},
+			sender: common.Address{1},
+			nonce:  4,
+		},
+		&dummyScramblerEntry{
+			hash:   common.Hash{5},
+			sender: common.Address{1},
+			nonce:  5,
+		},
+	}
+
+	shuffleEntries(entries)
+	entries = scrambleTransactions(42, entries)
+	for i := range entries {
+		for j := i + 1; j < len(entries); j++ {
+			if entries[i].Nonce() > entries[j].Nonce() {
+				t.Errorf("incorrect nonce order %d must be before %d", entries[j].Nonce(), entries[i].Nonce())
+			}
+		}
+	}
+}
+
+func TestTxScrambler_SortTransactionsWithSameSender_SortsByGasIfNonceIsSame(t *testing.T) {
+	entries := []ScramblerEntry{
+		&dummyScramblerEntry{
+			hash:     common.Hash{1},
+			sender:   common.Address{1},
+			nonce:    1,
+			gasPrice: big.NewInt(1),
+		},
+		&dummyScramblerEntry{
+			hash:     common.Hash{2},
+			sender:   common.Address{1},
+			nonce:    1,
+			gasPrice: big.NewInt(2),
+		},
+		&dummyScramblerEntry{
+			hash:     common.Hash{3},
+			sender:   common.Address{1},
+			nonce:    1,
+			gasPrice: big.NewInt(3),
+		},
+		&dummyScramblerEntry{
+			hash:     common.Hash{4},
+			sender:   common.Address{1},
+			nonce:    1,
+			gasPrice: big.NewInt(4),
+		},
+	}
+
+	shuffleEntries(entries)
+	entries = scrambleTransactions(42, entries)
+	for i := range entries {
+		for j := i + 1; j < len(entries); j++ {
+			if entries[i].GasPrice().Uint64() < entries[j].GasPrice().Uint64() {
+				t.Errorf("incorrect gas price order %d must be before %d", entries[i].GasPrice(), entries[j].GasPrice())
+			}
+		}
+	}
+}
+
+func TestTxScrambler_SortTransactionsWithSameSender_SortsByHashIfNonceAndGasIsSame(t *testing.T) {
+	entries := []ScramblerEntry{
```

Comment:
nit: you can reduce the details in the test by using default senders, nonces, and gas prices, which makes them all 0 -- which is a valid value for all of them.